src/transformers/modeling_utils.py

```diff
@@ -1665,17 +1747,22 @@ def tie_encoder_to_decoder_recursively(
             decoder_pointer: nn.Module,
             encoder_pointer: nn.Module,
             module_name: str,
+            base_encoder_name: str,
             uninitialized_encoder_weights: List[str],
             depth=0,
+            total_decoder_name="",
+            total_encoder_name="",
         ):
             assert isinstance(decoder_pointer, nn.Module) and isinstance(
                 encoder_pointer, nn.Module
             ), f"{decoder_pointer} and {encoder_pointer} have to be of type nn.Module"
             if hasattr(decoder_pointer, "weight"):
                 assert hasattr(encoder_pointer, "weight")
                 encoder_pointer.weight = decoder_pointer.weight
+                tied_weights.append(f"{base_encoder_name}{total_encoder_name}.weight")
```

Comment:
(not sure at all) but should there be a dot here between the names? 

```suggestion
                tied_weights.append(f"{base_encoder_name}.{total_encoder_name}.weight")
```

Reply:
No, the encode already has the leading dot from the way the recursive calls are made.

Forcing it here means adding extra logic in the recursive descent.
I can do it to make the code more readable (but in general in such complex code I don't like adding too many ifs especially on dependant varibles in recursive calls)

Reply:
Agreed - I'd rather no if statements